assets/src/edit-story/output/page.js

```diff
@@ -57,6 +48,18 @@ function OutputPage({ page, autoAdvance, defaultPageDuration }) {
         ...generatePatternStyles(backgroundColor),
       };
 
+  const animationDuration = getTotalDuration({ animations }) / 1000;
+  // If the page doesn't have media, take either the animations time or the configured default duration time.
+  const nonMediaPageDuration = Math.max(
+    animationDuration || 0,
+    defaultPageDuration
+  );
+  // If we have media, take the media time for advancement time and ignore the default,
+  // but still consider animation time as the minimum, too.
+  const longestMediaElement = getLongestMediaElement(
+    elements,
+    animationDuration || 1
+  );
```

Comment:
Hm, can you confirm, that this is the result?

| Video | Animation | Default | Actual page advancement time |
|-|-|-|-|
| 1s | 0s | 4s | 1s |
| 0s | 2s | 4s | 4s |
| 1s | 2s | 4s | 2s |

I don't find this intuitive, but any combination is pretty weird in my opinion anyway, so I'm not really challenging this, just asking to confirm...

Reply:
Yep, this is the intended result as per current logic.

The origin issue doesn't mention animations, just a change for how it behaves with media, however, it shouldn't advance before the animations stop either, so 🤷 

Let's wait for @o-fernandez to confirm before moving forward.

Reply:
Perhaps ping on Slack just in case

Reply:
I think that this works well most of the time. 

One caveat, if there's an animation that delays the start of a video, then the calculation for video time should include that. For example, video length is 1s, but we apply an animation so that the video fades in and starts playing after 2s, then the video won't finish playing until 3s from the page start time. That should be factored in. So it's not really just video length, but time until video plays once through. 

@miina @swissspidy @barklund @choumx thoughts on this? I think it's an edge case, but if y'all agree I can add that scenario to the spec and account for it.

Reply:
Since I'll be on time off for two weeks then would be great if we could create a new issue for that edge case (unless anyone else has capacity to pick it up meanwhile, moving this issue to QA meanwhile as it addresses the original issue).

Reply:
@o-fernandez, let's file that as a follow-up ticket. I actually think we have a bug with videos with animation - at least it doesn't work at all locally for me?

Reply:
> So it's not really just video length, but time until video plays once through.

That‘s how it‘s currently implemented.